op-service/sources/engine_client.go

```diff
@@ -58,7 +58,12 @@ func (s *EngineClient) ForkchoiceUpdate(ctx context.Context, fc *eth.ForkchoiceS
 	fcCtx, cancel := context.WithTimeout(ctx, time.Second*5)
 	defer cancel()
 	var result eth.ForkchoiceUpdatedResult
-	err := s.client.CallContext(fcCtx, &result, "engine_forkchoiceUpdatedV3", fc, attributes)
+	engineAPIVersion := rollup.EngineAPIV3
+	if attributes != nil {
+		engineAPIVersion = s.rollupCfg.ForkchoiceUpdatedVersion(uint64(attributes.Timestamp))
+	}
+	method := fmt.Sprintf("engine_forkchoiceUpdated%s", engineAPIVersion)
```

Comment:
fmt.Sprintf is a step backwards. We should return the method string from the version function. Ideally the FCU Version function should also take the attributes.

Reply:
Changed engine api version methods to return full method strings & take payload attr.